File: Talleres RESFTUL JAVA DOTNET/Dotnet/API-DOTNET-BANK/Application/DTOs/Transaction/TransactionDto.cs
namespace API_DOTNET_BANK.Application.DTOs.Transaction
{
    public class TransactionDto
    {
        public int Id { get; set; }
        public int TransactionType { get; set; }
        public decimal Amount { get; set; }
        public string? Description { get; set; }
        public int AccountId { get; set; }
        public string AccountNumber { get; set; } = default!;
        public int? DestinationAccountId { get; set; }
        public string? DestinationAccountNumber { get; set; }
        public DateTime CreatedAt { get; set; }
    }

    public class CreateDepositDto
    {
        public int AccountId { get; set; }
        public decimal Amount { get; set; }
        public string? Description { get; set; }
    }

    public class CreateWithdrawalDto
    {
        public int AccountId { get; set; }
        public decimal Amount { get; set; }
        public string? Description { get; set; }
    }

    public class CreateTransferDto
    {
        public int SourceAccountId { get; set; }
        public int DestinationAccountId { get; set; }
        public decimal Amount { get; set; }
        public string? Description { get; set; }
    }
}

File: Talleres RESFTUL JAVA DOTNET/API_GOOGLE_MAPS_REST_JAVA_GR-03/04. SERVIDOR/api-java-bank/src/main/java/com/allpasoft/apijavabank/application/dto/TransactionDto.java
package com.allpasoft.apijavabank.application.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TransactionDto {
    private Long id;
    private Integer transactionType;
    private BigDecimal amount;
    private String description;
    private Long accountId;
    private String accountNumber;
    private Long destinationAccountId;
    private String destinationAccountNumber;
    private LocalDateTime createdAt;
}
